src/speakers/extraction.py

```diff
@@ -329,3 +332,312 @@ def get_speakers_in_bold(doc):
         if speaker not in speakers:
             speakers.append(speaker)
     return speakers
+
+
+def get_lines(doc):
+    doc=fitz.open(doc)
+    pages=_get_main_pages(doc)
+    text = "\n".join([page.text for page in pages])
+    text = text.replace("Pvt.", "Pvt")
+    text = text.replace(":",":\n")
+    lines = [l.strip() for l in text.splitlines() if l.strip()]
+    return lines
+
+
+def check_last_char(names):
+    count=0
+    names_temp=[]
+    for name in names:
+        if name[-1]==':':
+            count=count+1
+    names_temp=names
+    names=[]
+    for name in names_temp:
+        if count/len(names_temp)>0.40:
+            if name[-1]==':':
+                names.append(name)
+        elif name[-1]!=':' and name[-1]!='.':
+            names.append(name)
+    return names
+
+
+def remove_last_char_if_colon(word):
+    if word[-1]==':':
+        word=word[0:len(word)-1]
+    return word
+
+
+def check_dict(names):
+    names_temp=names
+    names=[]
+    possible_names_in_dictionary=['Shah','Moderator','moderator','Raj','Participant']
+    for name in names_temp:
+        count=0
+        for word in name.split():
+            word=remove_last_char_if_colon(word)
+            if dictionary.check(word) and word not in possible_names_in_dictionary:
+                count=count+1
+            if name=='KFin':
+                count=count+1
+        if count/len(name.split())<=0.50:
+            names.append(name)
+    return names
+
+
+def check_repetations(names):
+    names_temp=names
+    for name_outer in names:
+        for name_inner in names:
+            if name_outer.find(name_inner)!=-1 and len(name_outer)>len(name_inner):
+                names_temp.remove(name_inner)
+    names=names_temp
+    return(names)
+
+
+def get_speaker_names(doc):
+    lines = get_lines(doc)
+    char_not_in_names=[',','-','&','?']
+    names=[]
+    converse=[]
+    for line in lines:
+        flag=0
+        words=line.split()
+        if len(words)<=5 or (words[0]=="Moderator" or words[0]=='Moderator:'): 
+            if len(words)==1 and len(words[0])<4:
+                flag=1
+            for word in words:
+                if not word[0].isupper():
+                    flag=1
+                for letter in word:
+                    if letter.isnumeric() or letter in char_not_in_names:
+                        flag=1
+            if flag==0:
+                if line not in names:
+                    names.append(line)
+    names=check_dict(names)
+    names=check_last_char(names)
+    names=check_repetations(names)
+    for line in lines:
+        if line in names:
+            converse.append(line)
```

Comment:
Converse here has the repeated names of people in the order they speak. Not sure if the variable name converse captures this.